android-lib/src/main/java/com/kinvey/android/UuidFactory.java

```diff
@@ -88,28 +89,31 @@ public UuidFactory(Context context) {
                             if (androidId != null && !"9774d56d682e549c".equals(androidId)) {
                                 uuid = UUID.nameUUIDFromBytes(androidId.getBytes("utf8"));
                             } else {
-                                final String deviceId = ((TelephonyManager) context
-                                        .getSystemService(Context.TELEPHONY_SERVICE))
-                                        .getDeviceId();
-                                uuid = deviceId != null ? UUID.nameUUIDFromBytes(deviceId.getBytes("utf8"))
-                                        : UUID.randomUUID();
+                                String deviceId = phoneDeviceId(context);
+                                uuid = deviceId != null ? UUID.nameUUIDFromBytes(deviceId.getBytes("utf8")) : UUID.randomUUID();
                             }
                         } catch (UnsupportedEncodingException e) {
                             throw new RuntimeException(e);
                         }
-
                         // Write the value out to the prefs file
                         prefs.edit().putString(PREFS_DEVICE_ID, uuid.toString()).commit();
-
                     }
-
                 }
             }
         }
-
     }
 
-
+    private String phoneDeviceId(Context context) {
```

Comment:
@b-stolyarov How are these changes for the device Id related to the Android Lint task?

Reply:
This changes is to fix error in `Lint` report, that fails build process.